### DISCUSSION TITLE
Porting to native ESP-IDF

### DISCUSSION THREAD

**Discussion by hiperiondev** [ElPercha/PLsi#60] [General]
Congratulations and thank you very much for this project.
It really seems to me to be a very interesting contribution.

Since I have serious differences with Arduino and I prefer to use pure C for embedded projects I started a port to the native ESP-IDF.
For now there is only access to disk and ladder (I think that today I end the physical access I / O).

I have made minor modifications especially to adapt it to the IDF, but there is still a lot to do.

The project here: https://github.com/hiperiondev/esp32-plc/

OT: El Percha? Argentino?

**Comment by hiperiondev** [comment]
There are al most two related interesting project:
https://github.com/61131/echidna
https://github.com/mprymek/PeaLC

**Comment by ElPercha** [comment]
Hi!
I'm glad you like it :)
I'd also love to have the arduino.h out of the project. But: "my skills" * "my available time* give me a low score. 
For this reason, I decided to go with Arduino: to speed up.

I really appreciate that you are using your time to port it to the native ESP-IDF. Thank you.
I'd love to help you on the process, the easiest thing for me would be to support testing it in the hardware (paid job is consuming a lot of energy)

OT: Yes! I'm from Argentina as well. Send me a message if you want to have a talk funnyautomation@gmail.com

**Comment by hiperiondev** [comment]
Te mandé un correo
